controllers/replication-controller/persistentvolumeclaim_controller.go

```diff
@@ -261,8 +291,84 @@ func (r *PersistentVolumeClaimReconciler) SetupWithManager(mgr ctrl.Manager, lim
 }
 
 func pvcProtectionIsComplete() predicate.Predicate {
-	return newPredicateFuncs(func(meta client.Object) bool {
-		a := getAnnotations(meta)
+	return predicate.NewPredicateFuncs(func(meta client.Object) bool {
+		a := meta.GetAnnotations()
 		return a != nil && a[controller.PVCProtectionComplete] == "yes"
 	})
 }
+
+func updatePVCAnnotations(pvc *v1.PersistentVolumeClaim, remoteClusterID string, pv *v1.PersistentVolume) {
```

Comment:
this function does multiple things other than updating the annotations for the input pvc object- updates the lables, metadata, spec params. Either rename this function to an appropriate name or separate the functionalities;

Reply:
Updated the name